runtime/marshal_httpbody.go

```diff
@@ -0,0 +1,74 @@
+package runtime
```

Comment:
Do you think you could add a test for this?

Reply:
yea sure, but let's finish the other discussion first so I don't have to change things too much.